### PR TITLE
Better theme support for FormField.

### LINKED ISSUE #2765: FormField is hard to customize
### Expected Behavior
FormField is fully customizable

### Actual Behavior
FormField is not passing `className` down to the container, to it is impossible to make `styled(FormField)` work.

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/n1k6kl413l

### Steps to Reproduce

1. Open sandbox link
2. notice the custom formfield overrides are not working

### Your Environment

- Grommet version: latest
- Browser Name and version: any
- Operating System and version (desktop or mobile): any

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?

Changes FormField to allow custom theme.

#### Where should the reviewer start?
FormField.js

#### What testing has been done on this PR?
added test for extend and add storybook story
#### How should this be manually tested?
open formfield stories and make sure it looks the same as https://storybook.grommet.io.
plus look at the new custom theme story to validate the new entries are supported.

#### What are the relevant issues?
Closes #2765 

#### Do the grommet docs need to be updated?
yes, we need theme documentation for formfield!!

#### Should this PR be mentioned in the release notes?
yes

#### Is this change backwards compatible or is it a breaking change?
backwards

### CODE REVIEW

File: src/js/components/FormField/formfield.stories.js [hunk 1]
```diff
@@ -133,10 +134,42 @@ const FormFieldHelpError = props => (
...
+
 storiesOf('FormField', module)
   .add('TextInput', () => <FormFieldTextInput />)
   .add('TextArea', () => <FormFieldTextArea />)
   .add('Select', () => <FormFieldSelect />)
   .add('CheckBox', () => <FormFieldCheckBox />)
   .add('Toggle', () => <FormFieldToggle />)
-  .add('Help and error', () => <FormFieldHelpError />);
+  .add('Help and error', () => <FormFieldHelpError />)
+  .add('Custom Theme', () => <CustomFormField />);
```
Comment (ShimiSun) [inline src/js/components/FormField/formfield.stories.js]: Thanks for adding a story!
I noticed these warnings when running it:

File: src/js/themes/base.js [hunk 2]
```diff
@@ -487,19 +487,33 @@ export const generate = (baseSpacing = 24, scale = 6) => {
           },
         },
       },
+      content: {
```
Comment (ericsoderberghp) [inline src/js/themes/base.js]: I think we should add a comment above `formField` indicating that these are setting grommet properties, not CSS directly.

Comment (alansouzati) [reply src/js/themes/base.js]: hum. i can definitely add a comment for that. but our theme object never assumed it was css only. take Select as an example, we use the theme to allow a custom component to use for the TextInput. we have other examples.

Although I agree most of the theme entries affect css directly, it is not 100% the case. So, I'm not sure if we should add a comment for all exceptions there.

File: src/js/components/FormField/formfield.stories.js (hunk 1 above)
Comment (alansouzati) [reply src/js/components/FormField/formfield.stories.js]: let me take a look...

**Pull request merged**